### ISSUE TITLE
Upgrade to latest create-react-app version

### ISSUE THREAD

**Issue by 4ian** [4ian/GDevelop#1470]
We should upgrade to the latest version of create-react-app.

Things to do:

- [ ] Check the release notes to see if there is any breaking changes
- [ ] Do the upgrade in package.json
- [ ] Check that the web-app still works (localhost:3000)
- [ ] Check that the standalone Electron app still works
- [ ] Verify that there is no warning/issues.
- [ ] Verify that `npm run storybook` still works. It has already created issues in the past. It must continue to work.

**Comment by piyushpalawat99** [comment]
Can I work on this issue?

**Comment by 4ian** [comment]
Sure! Feel free to go ahead :)

**Comment by piyushpalawat99** [comment]
Can you tell me why 

> Check the release notes to see if there is any breaking changes

is in the "Things to do" list and how would the breaking changes affect the app?

**Comment by 4ian** [comment]
There can be changes in create-react-app that make the app unable to start, or some configuration files to be changed, etc..
Usually as long as the app is starting (web-app and electron) (and storybook too) it should be ok - create-react-app is doing the bundling of the app but not used at runtime.

**Comment by piyushpalawat99** [comment]
Noted. Thanks.

**Comment by 4ian** [comment]
In other words, this might be a very simple upgrade or some unexpected difficulties can arise  - we'll see! :)

**Comment by arteevraina** [comment]
I want to work on this issue.

**Comment by arteevraina** [comment]
There are multiple json files in the project. Which json file we have to update in order to make the changes? @4ian

**Comment by piyushpalawat99** [comment]
@arteevraina I'm already working on this issue and will be sending a PR for the same soon.

**Comment by piyushpalawat99** [comment]
And just for your info @arteevraina , the file that has to be changed seems to be this one: https://github.com/4ian/GDevelop/blob/master/newIDE/app/package.json

But, I believe Florian can confirm this once he takes a look at the conversation
I've already committed changes and am running app to ensure that everything runs smoothly. Would be great if you could take a look at some other issues.

Cheers,
Piyush

**Comment by 4ian** [comment]
That's this package.json file - but looking at the content of the files, you could have deduced this, in theory no need for me to confirm anything here ;)

It's fine to work on same issue is you want to train, though obviously only one PR will be merged (well, depends, someone can find an issue that went unnoticed by someone else :))

**Comment by piyushpalawat99** [comment]
When trying to build the web-app after setting `"react-scripts": "3.4.0"` in Ubuntu 18.04 LTS, the following error appears when running npm start:

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gdevelop-ide@0.1.0 start: `npm run import-resources && npm run make-version-metadata && react-scripts start`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the gdevelop-ide@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/piyush/.npm/_logs/2020-02-29T09_03_44_085Z-debug.log
```
I tried installing and running it twice but to no avail. Is there anything else I can do?

**Comment by piyushpalawat99** [comment]
After the above error, I changed the react-scripts version to what it was before and everything works fine now, so the error seems to be due to a change in the react-scripts version.

Following are the breaking changes I found related to react-scripts:
Fix RegExp from navigateFallbackBlacklist (workbox)
Update url-loader to 2.0.1
A number of breaking changes in create-react-app v3.0.1: https://github.com/facebook/create-react-app/releases?after=babel-preset-react-app%409.0.1

**Comment by piyushpalawat99** [comment]
@4ian can you help me with this?

**Comment by 4ian** [comment]
Hard to say, the logs are not giving any useful information. You'll have to dig deeper to see what's failing.

> The web-app is able to run in create-react-app v2.1.8 though, should I change it to this version and send a PR?

Everything must work ;)

**Comment by piyushpalawat99** [comment]
Will let you know if I'm able to find the reason for failing and in the meantime, will test the different versions of app and storybook to make sure everything's fine while running with v2.1.8
Thanks

**Comment by NilayMajorwar** [comment]
Hey, I tried this and I think I got it to work with the latest version of `react-scripts`.

There was an issue with NPM dependency management - `@storybook/react` required a different version of `webpack` than react-scripts, and `npm install` disregarded the `react-scripts` webpack version dependency due to some reason. Had to bump up webpack separately in `package.json` to resolve that. Yarn worked fine, without any dependency issue.
